### ISSUE TITLE
"Instant and permanent friends"

### ISSUE THREAD

**Issue by luchaos** [RetroAchievements/RAWeb#813]
### Discussed in https://github.com/RetroAchievements/RAWeb/discussions/691

<div type='discussions-op-text'>

<sup>Originally posted by **JuliaSalsa** February 17, 2018</sup>
When you are requesting a friend, they are automathically added to friend leaderboard, from which you can't remove them.</div>

**Comment by Flashfire42** [comment]
This is the same site where blocking someone adds them as a friend. V2 should fix this

**Comment by televandalist** [comment]
The code for blocking someone is a literal copy-paste of the code for adding them. At the very least the button should be removed.

**Comment by Jamiras** [comment]
It's not 100% copy-paste. There's one character difference. When adding a friend, the database entry is set to "1". When blocking a user, it's set to "-1". I think the problem is there's a lot of code that just looks to see if there's an entry without checking the value.

`confirmFriend`:
https://github.com/RetroAchievements/RAWeb/blob/fc4b6a9034694e97a53e02cb2a5c56d700c5c57a/lib/database/friend.php#L178

`blockFriend`:
https://github.com/RetroAchievements/RAWeb/blob/fc4b6a9034694e97a53e02cb2a5c56d700c5c57a/lib/database/friend.php#L212

Note that this is a separate issue from the original topic, which is that there's no way to unfriend someone.

**Comment by ScottFromDerby** [comment]
> This is the same site where blocking someone adds them as a friend. V2 should fix this

lol uh shit sorry, my bad

**Comment by Jamiras** [comment]
> Note that this is a separate issue from the original topic, which is that there's no way to unfriend someone.

It looks like the "Remove Friend" and "Unblock User" functionality is working fine.
